### ISSUE TITLE
Complete bipartite graph bug

### ISSUE THREAD

**Issue by andre-ve** [sofiat-olaosebikan/hopcroftkarp#1]
If I use a complete bipartite graph {'S3': set(['A', 'C', 'B']), 'S2': set(['A', 'C', 'B']), 'S1': set(['A', 'C', 'B'])}
I get {'A': 'S2', 'S2': 'A', 'S1': 'C', 'C': 'S1'} as a result, which is not a maximum matching. Maximum matching is clearly with three edges.

**Comment by sofiatolaosebikan** [comment]
Thank you very much, for pointing out the bug. 

I observed that the choice of the key you used in your dictionary was the problem. The following structure of bipartite graph {'a': set([1, 2, 3]), 'b': set([1, 2, 3]), 'c': set([1, 2, 3])} produced {1: 'b', 2: 'a', 3: 'c', 'c': 3, 'b': 1, 'a': 2}, which is a maximum matching. 

However, I have fixed the bug, and now the structure of your bipartite graph {'S3': set(['A', 'C', 'B']), 'S2': set(['A', 'C', 'B']), 'S1': set(['A', 'C', 'B'])} produced {'A': 'S3', 'C': 'S2', 'B': 'S1', 'S3': 'A', 'S2': 'C', 'S1': 'B'}, which is a maximum matching.
